### ISSUE TITLE
New Lasso tool in Segment editor

### ISSUE THREAD

**Issue by lassoan** [Slicer/Slicer#5606]
## Is your feature request related to a problem? Please describe.

Sometimes it would be useful to edit the contour drawn using Scissors effect before cutting, or cutting with a polyline similarly to Draw effect.

Sometimes it would be useful for Draw effect to have modes similarly to Scissors effect (fill/erase, different shapes and constraints).

Users often don't realize how versatile Scissors tool is and that it can be used for all kinds of drawings.

## Describe the solution you'd like

Merging Draw and Scissors into a new "Lasso" effect could be an answer to all the above needs.

The Scissors effect could have a mode where drawing does not complete immediately when the mouse is released but you could define a curve (freehand, polyline, or spline) by clicking in the image, and in the end finalize on double-click or right-click. We should also allow moving of control points before finalizing.

We now have a good infrastructure for this: markups closed contours. The Lasso effect could use markups similarly how Surface cut effect uses markups fiducials.

## Describe alternatives you've considered

Educate users with improving documentation about capabilities of Scissors effect. It might help somewhat, but would not be a full solution.

## Additional context

See discussion in https://github.com/Slicer/Slicer/issues/5605.

**Comment by pieper** [comment]
Yes, it would be great to somehow unify the concepts in the scissors/draw with the markups.  That is, it would be great to be able to construct the scissors region, preview the result, and edit before finalizing.  So the initial draw gesture would be 'frozen' like a freehand markup and you could change any of the parameters like the symmetric distance interactively and see the result; we could even use a draggable plane markup to set the parameter more like the dynamic modeler.
